include/sdsl/int_vector_buffer.hpp

```diff
@@ -266,50 +248,54 @@ class int_vector_buffer
             return (m_buffersize*width())/8;
         }
 
-        //! Set the buffersize in bytes
-        void buffersize(uint64_t buffersize)
-        {
-            if (0ULL == buffersize)
-                buffersize = 8;
-            write_block();
-            if (0==(buffersize*8)%width()) {
-                m_buffersize = buffersize*8/width(); // m_buffersize might not be multiple of 8, but m_buffersize*width() is.
-            } else {
-                uint64_t element_buffersize = (buffersize*8)/width()+1; // one more element than fits into given buffersize in byte
-                m_buffersize = element_buffersize+7 - (element_buffersize+7)%8; // take next multiple of 8
-            }
-            m_buffer = int_vector<t_width>(m_buffersize, 0, width());
-            if (0!=m_buffersize) read_block(0);
-        }
-
         //! Returns whether state of underlying streams are good
         bool good()
         {
-            return m_ifile.good() and m_ofile.good();
+            return m_ifile->good() and (!m_ofile->is_open() || m_ofile->good());
         }
 
         //! Returns whether underlying streams are currently associated to a file
         bool is_open()
         {
-            return m_ifile.is_open() and m_ofile.is_open();
+            return m_ifile->is_open();
```

Comment:
why are you only checking `m_ifile` here? shouldn't it depend on the mode?

Reply:
`m_ifile` should always be open if the stream is active.
`m_ofile`, however, may be closed even for an initialized stream (when it's open for reading only).